server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestResizeHandler.java

```diff
@@ -48,17 +48,22 @@ public final RestChannelConsumer prepareRequest(final RestRequest request, final
         final ResizeRequest resizeRequest = new ResizeRequest(request.param("target"), request.param("index"));
         resizeRequest.setResizeType(getResizeType());
         final String rawCopySettings = request.param("copy_settings");
-        final boolean copySettings;
+        final Boolean copySettings;
         if (rawCopySettings == null) {
             copySettings = resizeRequest.getCopySettings();
+        } else if (rawCopySettings.isEmpty()) {
+            copySettings = true;
```

Comment:
Is this right? It seems like this would make the `copySettings` default `true` for the rest layer but false for the transport layer?

Could we not let `TransportResizeAction` be the only thing that handles the `null` case? So to change the default from `false` to `true` in 8.x we only need to change it in that one place?

Reply:
That's not what this means, this is the non-null but empty case for `copy_settings`. This handles the case when a user invokes `/source/_shrink/target?copy_settings`; we have a convention in our APIs that `?parameter` without `=true` or `=false` means `true` (e.g., `?pretty).

I am also okay with some duplication for temporary code that is scoped. This code in master is going to immediately be modified so that we default to `true`.

Reply:
Also, I prefer to fail early on the REST layer rather than waiting for a round-trip to the master and resulting in a remote transport exception.

Reply:
> Also, I prefer to fail early on the REST layer rather than waiting for a round-trip to the master and resulting in a remote transport exception.

That was my point with https://github.com/elastic/elasticsearch/pull/30404/files/6c0a153bc037df67dcca2b036dcb367e3f084081#r186675743 of doing the validation in the `setCopySettings` method and throwing an `IllegalArgumentException` there if the value passed in is false. It means we only have the validation in one place and the exception is thrown at the point the value is set (whether from the REST layer or the transport client).

Reply:
Right, but then we can not provide a clear error message. On the REST layer we want to tell the user that `copy_settings` can not be false, on the transport layer we want to tell the user that `copySettings` can not be false. I am fine moving the transport validation to the setter but I think the REST validation should remain?

Reply:
ok, agreed

Reply:
@colings86 I pushed.